File: src/main/java/me/miki/shindo/management/mods/SimpleHUDMod.java
package me.miki.shindo.management.mods;

import me.miki.shindo.Shindo;
import me.miki.shindo.management.language.TranslateText;
import me.miki.shindo.management.nanovg.NanoVGManager;
import me.miki.shindo.management.nanovg.font.Fonts;

public class SimpleHUDMod extends HUDMod {

    public SimpleHUDMod(TranslateText nameTranslate, TranslateText descriptionText) {
        super(nameTranslate, descriptionText);
    }

    public SimpleHUDMod(TranslateText nameTranslate, TranslateText descriptionText, String alias) {
        super(nameTranslate, descriptionText, alias);
    }

    public void draw() {

        Shindo instance = Shindo.getInstance();
        NanoVGManager nvg = instance.getNanoVGManager();
        boolean hasIcon = getIcon() != null;
        float addX = hasIcon ? this.getTextWidth(getIcon(), 9.5F, Fonts.LEGACYICON) + 4 : 0;

        if (getText() != null) {
            nvg.setupAndDraw(() -> {

                float bgWidth = (this.getTextWidth(this.getText(), 9, getHudFont(1)) + 10) + addX;

                this.drawBackground(bgWidth, 18);
                this.drawText(this.getText(), 5.5F + addX, 5.5F, 9, getHudFont(1));

                if (hasIcon) {
                    this.drawText(getIcon(), 5.5F, 4F, 10.4F, Fonts.LEGACYICON);
                }

                this.setWidth((int) bgWidth);
                this.setHeight(18);
            });
        }
    }

    public String getText() {
        return null;
    }

    public String getIcon() {
        return null;
    }
}

File: src/main/kotlin/me/miki/shindo/management/mods/SimpleHUDMod.kt
package me.miki.shindo.management.mods

import me.miki.shindo.Shindo
import me.miki.shindo.management.language.TranslateText
import me.miki.shindo.management.nanovg.font.Fonts


open class SimpleHUDMod : HUDMod {

    constructor(nameTranslate: TranslateText, descriptionText: TranslateText, icon: String) : super(
        nameTranslate,
        descriptionText,
        icon
    )

    constructor(nameTranslate: TranslateText, descriptionText: TranslateText, icon: String, alias: String) : super(
        nameTranslate,
        descriptionText,
        icon,
        alias
    )

    fun draw() {
        val instance = Shindo.getInstance()
        instance.nanoVGManager
        val icon = getIcon()
        val hasIcon = icon != null
        val addX = if (hasIcon) (getTextWidth(icon ?: "", 9.5f, Fonts.LEGACYICON)!! + 4) else 0f

        val text = getText()
        if (text != null) {
            val bgWidth = getTextWidth(getText()!!, 9f, getHudFont(1))!! + 10 + addX

            this.drawBackground(bgWidth, 18f)
            this.drawText(getText()!!, 5.5f + addX, 5.5f, 9f, getHudFont(1))

            if (hasIcon) {
                this.drawText(getIcon()!!, 5.5f, 4f, 10.4f, Fonts.LEGACYICON)
            }

            setWidth(bgWidth.toInt())
            setHeight(18)
        }
    }

    open fun getText(): String? {
        return null
    }

}
